### PR TITLE
Use GIT_PROGRAM for internal git commands

### PR DESCRIPTION
**Pull request by zimme**
So here's a quick attempt to get yadm to play nice with hub.
There are probably things that needs changing to work properly
for all cases but I thought I should at least try and give you a hand.

This will also add an exception for the clone command when hub is used.
The exception is that we forward all arguments for clone when hub is
used instead of git.

### CODE REVIEW

File: yadm [hunk 1]
```diff
@@ -155,22 +155,30 @@ function clone() {
 
   #; add the specified remote, and configure the repo to track origin/master
   debug "Adding remote to new repo"
-  git remote add origin "$1"
+  if [$GIT_PROGRAM = "hub"] ; then
+    "$GIT_PROGRAM" remote add origin "$@"
+  else
+    "$GIT_PROGRAM" remote add origin "$1"
+  fi
```
Comment (zimme) [inline yadm]: This part is where we check if we're running hub and if we do, send in all arguments that the user supplied to the clone command.
```
yadm clone -p zimme/dotfiles
// Internal call
hub remote add origin -p zimme/dotfiles
```

**Comment by zimme** [comment]
Also I made this pull request against the `exp` branch.

**Comment by TheLocehiliosan** [comment]
Thanks for the input. I'll look at this closer. It seems the major information is that the parameters for the `clone` and `remote add` commands are very similar, and possibly could be substituted for. Shouldn't be very difficult to handle.

**Comment by TheLocehiliosan** [comment]
This duplicates #30. This feature has been implemented in `dev` and will be released in version 1.06.

**Pull request closed without merging**